sklearn/cluster/tests/test_affinity_propagation.py

```diff
@@ -66,7 +67,14 @@ def test_affinity_propagation_predict():
     labels2 = af.predict(X)
     assert_array_equal(labels, labels2)
 
-
+    
+def test_affinity_propagation_convergence_warning():
+    # Test warning in AffinityPropagation.fit
+    af = AffinityPropagation(affinity="euclidean", max_iter=1)
+    assert_warns(ConvergenceWarning, af.fit, X)
+    assert_raises(ValueError, af.predict, X)
+    
```

Comment:
Could you please add a non-regression test that checks that a bad damping level can cause the convergence warning?